libtextsecure/message_receiver.js

```diff
@@ -661,19 +663,17 @@ MessageReceiver.prototype.extend({
         var sessionCipher = new libsignal.SessionCipher(textsecure.storage.protocol, address, options);
         console.log('retrying prekey whisper message');
         return this.decryptPreKeyWhisperMessage(ciphertext, sessionCipher, address).then(function(plaintext) {
-            var finalMessage = textsecure.protobuf.DataMessage.decode(plaintext);
-
-            var p = Promise.resolve();
-            if ((finalMessage.flags & textsecure.protobuf.DataMessage.Flags.END_SESSION)
-                    == textsecure.protobuf.DataMessage.Flags.END_SESSION &&
-                    finalMessage.sync !== null) {
-                    var number = address.getName();
-                    p = this.handleEndSession(number);
-            }
-
-            return p.then(function() {
-                return this.processDecrypted(finalMessage);
-            }.bind(this));
+            var now = Date.now();
+            var envelope = {
+                source: number,
+                sourceDevice: device,
+                timestamp: {
+                    toNumber: function() {
+                        return now;
```

Comment:
So, if I understand this correctly, upon entering a thread, the client pulls profiles, and if the identity keys in those profiles correspond to any un-decrypted messages (they would have had to predate the non-blocking SN changes), they will be decrypted and appear as "new" messages at the bottom of the thread. Is that right?

I'm not sure that's desirable - do we have the option of decrypting them at their original timestamp location? Presumably these messages have already been seen on the users corresponding mobile device.